zuul-core/src/main/java/com/netflix/zuul/message/http/HttpQueryParams.java

```diff
@@ -72,12 +78,17 @@ public static HttpQueryParams parse(String queryString) {
                 String name = s.substring(0, i);
                 String value = s.substring(i + 1);
 
-                try {
-                    name = URLDecoder.decode(name, "UTF-8");
-                    value = URLDecoder.decode(value, "UTF-8");
-                }
-                catch (Exception e) {
-                    // do nothing
+                if (!propertyFactory.getBooleanProperty(
+                    ZuulConstants.ZUUL_KEEP_ORIGINAL_QUERY_STRING_ENCODING,
+                    false).getValue()
+                ){
+                    try {
+                        name = URLDecoder.decode(name, "UTF-8");
```

Comment:
Hmm, I was going to comment on this, but it seems the issue is pre-existing.    The name / value decoding should be done as all or nothing, because otherwise an exception in the value decoder could cause tearing.   

Optional if you want to fix this.

Reply:
Based on what has been currently implemented I think it should be a per key/value. Cause in cases where the query string has only key the encoding decoding of the query string varies. I think we should leave it as is.